### PR TITLE
GraphQl-509 save address to address book

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
### Description (*)
Provided support of saving an address in an address book on quote address mutations.

### Fixed Issues (if relevant)
1. #509: `save_in_address_book` has no impact on Address Book

### CODE REVIEW

**Comment by lenaorobei** [comment]
@VitaliyBoyko please address small issues in static build. `testCodeStyle` failure is related to PHPDoc blocks and `testCodeMess` to `coupling between objects` and `cyclomatic complexity`. Thank you.

File: app/code/Magento/QuoteGraphQl/Model/Cart/SetBillingAddressOnCart.php [hunk 1]
```diff
@@ -88,16 +88,19 @@ public function execute(ContextInterface $context, CartInterface $cart, array $b
...
             $billingAddress = $this->quoteAddressFactory->createBasedOnCustomerAddress(
                 (int)$customerAddressId,
-                (int)$customer->getId()
+                $customerId
             );
         }
-
         $this->assignBillingAddressToCart->execute($cart, $billingAddress, $useForShipping);
+
+        if (!empty($addressInput) && !empty($addressInput['save_in_address_book']) && 0 !== $customerId) {
```
Comment (lenaorobei) [inline app/code/Magento/QuoteGraphQl/Model/Cart/SetBillingAddressOnCart.php]:
```suggestion
        if (0 !== $customerId && !empty($addressInput['save_in_address_book']))
```

File: app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressesOnCart.php [hunk 2]
```diff
@@ -73,16 +72,19 @@ public function execute(ContextInterface $context, CartInterface $cart, array $s
...
             $shippingAddress = $this->quoteAddressFactory->createBasedOnCustomerAddress(
                 (int)$customerAddressId,
-                (int)$customer->getId()
+                $customerId
             );
         }
-
         $this->assignShippingAddressToCart->execute($cart, $shippingAddress);
+
+        if (!empty($addressInput) && !empty($addressInput['save_in_address_book']) && 0 !== $customerId) {
```
Comment (lenaorobei) [inline app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressesOnCart.php]:
```suggestion
        if (0 !== $customerId && !empty($addressInput['save_in_address_book']))
```

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-5243](https://jira.corp.magento.com/browse/ENGCOM-5243) has been created to process this Pull Request

**Comment by naydav** [comment]
@VitaliyBoyko
Do you have some updates on this ticket? Just mainline was merged without any progress.
Thanks

**Comment by naydav** [comment]
Hello @VitaliyBoyko ,
The ticket was unassigned due to inactivity. Also, during this time, the mainline has changed very much and the code in the PR becomes outdated.
Please assign it again if you wish to continue.
Thanks!

**Pull request closed without merging**